Add calendar events endpoints (POST/GET) with DB migrations and UI enhancements

silas-mariosa opened the pull request:
[ ] post newEvent

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Novos Recursos**
  * Eventos do calendário agora têm campos de início e término; formulário suporta all-day e reset automático ao editar.
  * Registro de usuário inclui telefone e campo de aceite de termos (pré-definido como não marcado).
  * Construtor de formulários ganhou suporte a textarea e opção de desabilitar campos.

* **Testes**
  * Nova suíte de testes para rotas de calendário com verificação de criação de evento.

* **Chores**
  * Migração/ snapshot do banco atualizados; CI e dependências de desenvolvimento ajustadas.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mendesmariosatech committed:
Scalar openAPI (#46)

* with scalar openapi

* Update videos.ts

* create register open api

* login route

* ENV_VARIABLES

* Update auth.test.ts

* env

* all examples working

* test wip

* chore: test running

* working docs and server tests

* env error

* Update setup.ts

* wip

* format working branch

* fix tests

* small updates

* !! scaler not working

* scalar path not right

* bug: skip test

* address code review

---------

Co-authored-by: Alex Mendes <46525527+mendes-develop@users.noreply.github.com>
Co-authored-by: Silas Roberto Mariosa de Vasconcelos <113197484+silas-mariosa@users.noreply.github.com>

mendes-develop committed:
rename tables and drop old tables

- Start an events route
- create a generate id function

mendes-develop committed:
create schema from drizzle zod

mendes-develop committed:
scalar docs

mendes-develop committed:
next app should install vercel hono

mendes-develop committed:
improvements around testing server

mendes-develop committed:
node hono

mendes-develop committed:
Create index.test.ts

mendes-develop committed:
Integrate first creation test

mendes-develop committed:
date working

mendes-develop committed:
add middleware to event creation

mendes-develop committed:
error setup

mendes-develop committed:
test

mendes-develop committed:
tests almost passing

mendes-develop committed:
Update index.test.ts

mendes-develop committed:
failing test

mendes-develop committed:
fix tests and remove console.logs

mendes-develop committed:
add setTimout

mendes-develop committed:
attempt to fix error

mendes-develop committed:
test 2

mendes-develop committed:
revert mock

mendes-develop committed:
add real token

mendes-develop committed:
working locally

mendes-develop committed:
Update jest.yml

mendes-develop committed:
reorder

mendes-develop committed:
rename test files

mendes-develop committed:
better errors

mendes-develop committed:
typed variables, businessId in context

mendes-develop committed:
env variables type in one place

mendes-develop committed:
create business

mendes-develop committed:
type errors scalar docs and business route

mendes-develop committed:
rename and change script

mendes-develop committed:
bring business to the routes

mendes-develop committed:
Update schema.ts

mendes-develop committed:
require headers

mendes-develop committed:
create business

silas-mariosa committed:
MM-9 Calendar Builder (#43)

* chore: add CalendarBuilder add CalendarFull add CollapsableBuilder

* chore: adjusted collapseble component, adjusted layout CalendarColumnBuilder

* chore: add FullCalendar Lib, add location language

* chore: update calendar props

* Chore: add CRUD Calendar, add tag (not work well)

* chore: crud stage incomplete

* chore: add fullcalendar builder

* chore: resolve comment 1

* chore: adjusted CrotrolledForm

* chore: ControlledForm breack

* usar react hook form

* arrumar types

* remove unnecessary files

* more types

* Update sidebar.tsx

* update form builder

* reset DB

---------

Co-authored-by: Alex Mendes <46525527+mendes-develop@users.noreply.github.com>

mendes-develop committed:
MM-1 Video Platform (#45)

* sonner

* sonner on a complete and error call

* add sonner to Docs

* modified:   packages/hook-services/react-query/useLogin.ts

* add dashboard to docs

* Update useLogin.ts

* Update Dashboard.stories.tsx

* navigate

* Create index.ts

* move dashboard

* success local

* Update useLogin.ts

* Update Dashboard.tsx

* add emoji

* Add favicon and fix storybook build because of useRouter

* change images and copy

* block user from accessing private routes

* chore: primerra etapa da migracao

* Update VideoList.stories.tsx

* last changes

MiStefanoni committed:
MM-24 Renomear heading to hero (#47)

* Renomear heading to hero

* Trocar componente landingpage pra hero

---------

Co-authored-by: Alex Mendes <46525527+mendes-develop@users.noreply.github.com>

mendes-develop committed:
Merge branch 'dev' into MM-23_Server-Criar-Events,-Customer-e-Business-Table-e-renomear-Auth-

mendes-develop committed:
Merge branch 'MM-23_Server-Criar-Events,-Customer-e-Business-Table-e-renomear-Auth-' into MM-12_Server-POST-/calendar/events

mendes-develop committed:
remove logs

mendes-develop committed:
MM-23 server criar events, customer e business table e renomear auth  (#50)

* rename tables and drop old tables

- Start an events route
- create a generate id function

* address code rabbit changes

* Update login.ts

mendes-develop committed:
Merge branch 'dev' into MM-12_Server-POST-/calendar/events

mendes-develop committed:
Update calendar.test.ts

mendes-develop committed:
rm db and update schema

mendes-develop committed:
migrate

mendes-develop committed:
Update event.dto.ts

mendes-develop committed:
@coderabbitai review

silas-mariosa committed:
chore: adjust on get StarDay and EndDay on modal.

mendes-develop committed:
MM 12 server post /calendar/events (#51)

* rename tables and drop old tables

- Start an events route
- create a generate id function

* create schema from drizzle zod

* scalar docs

* next app should install vercel hono

* improvements around testing server

* node hono

* Create index.test.ts

* Integrate first creation test

* date working

* add middleware to event creation

* error setup

* test

* tests almost passing

* Update index.test.ts

* failing test

* fix tests and remove console.logs

* add setTimout

* attempt to fix error

* test 2

* revert mock

* add real token

* working locally

* Update jest.yml

* reorder

* rename test files

* better errors

* typed variables, businessId in context

* env variables type in one place

* create business

* type errors scalar docs and business route

* rename and change script

* bring business to the routes

* Update schema.ts

* require headers

* create business

* remove logs

* Update calendar.test.ts

* rm db and update schema

* migrate

* Update event.dto.ts

* @coderabbitai review

* chore: adjust on get StarDay and EndDay on modal.

* Revert "chore: adjust on get StarDay and EndDay on modal."

This reverts commit 3da128388720981d454fac0a3f6dab3b1b01dba3.

---------

Co-authored-by: silas-mariosa <silas.mariosa@hotmail.com>

MiStefanoni committed:
Michele/renomear heading to hero (#55)

* Renomear heading to hero

* Trocar componente landingpage pra hero

* Alex/release 0.1.2 (#53)

* Scalar openAPI (#46)

* with scalar openapi

* Update videos.ts

* create register open api

* login route

* ENV_VARIABLES

* Update auth.test.ts

* env

* all examples working

* test wip

* chore: test running

* working docs and server tests

* env error

* Update setup.ts

* wip

* format working branch

* fix tests

* small updates

* !! scaler not working

* scalar path not right

* bug: skip test

* address code review

---------

Co-authored-by: Alex Mendes <46525527+mendes-develop@users.noreply.github.com>
Co-authored-by: Silas Roberto Mariosa de Vasconcelos <113197484+silas-mariosa@users.noreply.github.com>

* MM-9 Calendar Builder (#43)

* chore: add CalendarBuilder add CalendarFull add CollapsableBuilder

* chore: adjusted collapseble component, adjusted layout CalendarColumnBuilder

* chore: add FullCalendar Lib, add location language

* chore: update calendar props

* Chore: add CRUD Calendar, add tag (not work well)

* chore: crud stage incomplete

* chore: add fullcalendar builder

* chore: resolve comment 1

* chore: adjusted CrotrolledForm

* chore: ControlledForm breack

* usar react hook form

* arrumar types

* remove unnecessary files

* more types

* Update sidebar.tsx

* update form builder

* reset DB

---------

Co-authored-by: Alex Mendes <46525527+mendes-develop@users.noreply.github.com>

* MM-1 Video Platform (#45)

* sonner

* sonner on a complete and error call

* add sonner to Docs

* modified:   packages/hook-services/react-query/useLogin.ts

* add dashboard to docs

* Update useLogin.ts

* Update Dashboard.stories.tsx

* navigate

* Create index.ts

* move dashboard

* success local

* Update useLogin.ts

* Update Dashboard.tsx

* add emoji

* Add favicon and fix storybook build because of useRouter

* change images and copy

* block user from accessing private routes

* chore: primerra etapa da migracao

* Update VideoList.stories.tsx

* last changes

* MM-24 Renomear heading to hero (#47)

* Renomear heading to hero

* Trocar componente landingpage pra hero

---------

Co-authored-by: Alex Mendes <46525527+mendes-develop@users.noreply.github.com>

* MM-23 server criar events, customer e business table e renomear auth  (#50)

* rename tables and drop old tables

- Start an events route
- create a generate id function

* address code rabbit changes

* Update login.ts

---------

Co-authored-by: mendesmariosatech <mendesmariosatech@gmail.com>
Co-authored-by: Silas Roberto Mariosa de Vasconcelos <113197484+silas-mariosa@users.noreply.github.com>
Co-authored-by: Michele Stefanoni <153405495+MiStefanoni@users.noreply.github.com>

* Renomear CoreServices para CardStrip

* Revert "merge with dev"

This reverts commit 8dc77996117466b73a4ff50dda275f5e61f9a33d, reversing
changes made to f777f878ab9c5f6f27b9b6d4216612fa8f1b78ba.

---------

Co-authored-by: Alex Mendes <46525527+mendes-develop@users.noreply.github.com>
Co-authored-by: mendesmariosatech <mendesmariosatech@gmail.com>
Co-authored-by: Silas Roberto Mariosa de Vasconcelos <113197484+silas-mariosa@users.noreply.github.com>

silas-mariosa committed:
chore: change postEvent change schema eventTable (url_invalid)

silas-mariosa committed:
chore: error drizzle generate

mendes-develop committed:
test backend in production (#59)

* test backend in production

* Update .coderabbit.yaml

* change PORT varibles

silas-mariosa committed:
Merge branch 'MM-12_Server-POST-/calendar/events' of https://github.com/mendesmariosatech/mmtech into MM-12_Server-POST-/calendar/events

silas-mariosa committed:
chore: post event

silas-mariosa committed:
Merge branch 'MM-12_Server-POST-/calendar/events' of https://github.com/mendesmariosatech/mmtech into MM-12_Server-POST-/calendar/events

silas-mariosa committed:
chore: add event/dto folder

silas-mariosa committed:
chore: change omit on schema event

silas-mariosa committed:
Merge branch 'dev' into MM-12_Server-POST-/calendar/events

mendes-develop commented on packages/server/drizzle.config.ts:
```diff
@@ -1,15 +1,15 @@
 import { defineConfig } from "drizzle-kit";
 
-config({ path: ".env" });
+console.log("TURSO_CONNECTION_URL:", process.env.TURSO_CONNECTION_URL);
+console.log("TURSO_AUTH_TOKEN:", process.env.TURSO_AUTH_TOKEN);
```
@silas-mariosa, concordo. Vamos remover

silas-mariosa replied:
Vou remover

mendes-develop commented on packages/server/src/routes/auth/register.ts:
```diff
@@ -83,13 +84,22 @@ export const registerHandler: AppRouteHandler<RegisterRoute> = async (c) => {
 		authId: newAuthUser.id,
 	});
 
+	const business = await Business.createBusiness({
+		name: name,
+		clientId: newClient!.id,
+		description: "My business",
+	});
```
Remover! O business nao eh pra ser criado aqui

silas-mariosa replied:
Removido

mendes-develop commented on packages/server/package.json:
```diff
@@ -20,6 +20,7 @@
     "@repo/zod-types": "*",
     "@scalar/hono-api-reference": "^0.5.156",
     "bcryptjs": "^2.4.3",
+    "cjs-loader": "^0.1.0",
```
remover

mendes-develop replied:
@silas-mariosa

silas-mariosa replied:
Removido

mendes-develop commented on package.json:
```diff
@@ -21,6 +21,7 @@
     "@biomejs/biome": "1.9.2",
     "@repo/eslint-config": "*",
     "@repo/typescript-config": "*",
+    "drizzle-kit": "0.24.2",
```
@silas-mariosa eu nao gosto muito da ideia de ter essa dependência aqui ja que ela só serve pro server. Sera que a pode tentar resolver isso no futuro?

mendes-develop commented on packages/server/src/routes/auth/register.ts:
```diff
@@ -67,10 +67,11 @@ export const registerHandler: AppRouteHandler<RegisterRoute> = async (c) => {
 
 	const Auth = new AuthTable(TURSO_CONNECTION_URL, TURSO_AUTH_TOKEN);
 	const Client = new ClientTable(TURSO_CONNECTION_URL, TURSO_AUTH_TOKEN);
+	const Business = new BusinessTable(TURSO_CONNECTION_URL, TURSO_AUTH_TOKEN);
```
@silas-mariosa, remover logica de criacao de Business aqui

silas-mariosa replied:
removido

mendes-develop commented on packages/server/src/routes/calendar/create-event.ts:
```diff
@@ -17,7 +17,8 @@ export const createEventSpec = createRoute({
 						description: z.string(),
 						date: z.string(),
-						time: z.string(),
+						startTime: z.string(),
+						endTime: z.string(),
```
@silas-mariosa, nessa aqui ele foi cirúrgico! Eu também acho legal mudar o frontend pra refletir isso

mendes-develop reviewed (changes requested):
Resolver conflitos, promover as mudanças que eu comentei, principalmente remover o business da rota de auth e fazer o update no Modal pra refleter as propriedades na Table de Eventos

silas-mariosa committed:
chore: remove business on auth

silas-mariosa commented on packages/server/src/routes/auth/register.ts:
```diff
@@ -67,10 +67,11 @@ export const registerHandler: AppRouteHandler<RegisterRoute> = async (c) => {
 
 	const Auth = new AuthTable(TURSO_CONNECTION_URL, TURSO_AUTH_TOKEN);
 	const Client = new ClientTable(TURSO_CONNECTION_URL, TURSO_AUTH_TOKEN);
+	const Business = new BusinessTable(TURSO_CONNECTION_URL, TURSO_AUTH_TOKEN);
```
foi removido

silas-mariosa committed:
Merge remote-tracking branch 'origin' into MM-12_Server-POST-/calendar/events

silas-mariosa commented on packages/server/drizzle.config.ts:
```diff
@@ -1,15 +1,15 @@
 	dbCredentials: {
-		url: process.env.TURSO_CONNECTION_URL,
-		authToken: process.env.TURSO_AUTH_TOKEN,
+		url: process.env.TURSO_CONNECTION_URL!,
+		authToken: process.env.TURSO_AUTH_TOKEN!,
```
vou remover

silas-mariosa commented on packages/server/src/routes/calendar/dto/event.dto.ts:
```diff
@@ -0,0 +1,14 @@
+		super(TURSO_CONNECTION_URL, TURSO_AUTH_TOKEN);
+	}
+
+	public async createEvent(args: InsertEvent) {
+		const [event] = await this.db.insert(eventTable).values(args).returning();
+
+		return event;
+	}
```
Os dados startTime e endTime podem ser null.

silas-mariosa committed:
chore: add adjutes to stay conform suggest

mendes-develop committed:
Add GET calendar route and enhance calendar events functionality

mendes-develop committed:
Merge main into calendar events branch - resolve conflicts

mendes-develop commented:
Addressed PR feedback by updating the title to better reflect the scope of changes and adding docstrings to improve code documentation coverage.

mendes-develop committed:
Add documentation to calendar routes and DTOs to improve code clarity

mendes-develop committed:
add value

mendes-develop committed:
Fix JSX import issue and add JWT algorithm parameter

mendes-develop committed:
Merge branch 'MM-12_Server-POST-/calendar/events' of https://github.com/mendesmariosatech/mmtech into MM-12_Server-POST-/calendar/events

mendes-develop committed:
Fix dbContext middleware type compatibility issue

mendes-develop committed:
Resolve merge conflicts in PR #63

mendes-develop committed:
Resolve merge conflicts in PR #63

mendes-develop committed:
Fix: Apply CodeRabbit suggestions to improve code quality and fix issues

- Add HS256 algorithm to JWT signing function
- Fix businessId typing as optional in JWT decoding
- Remove duplicate  in auth table schema
- Fix agreeTerms field to be notNull without default in schema
- Improve InsertEventSchema with proper nullable transforms for startTime/endTime
- Fix modal title logic to properly check for event existence
- Remove non-null assertion operator in delete button
- Add conditional rendering for DevTool in CalendarModal
- Add optional value property to checkbox config
- Fix checkbox checked state with proper fallback
- Add proper styling and disabled attribute to textarea input
- Remove duplicate dotenv import in app.ts
- Remove duplicate afterAll block in calendar tests
- Use .pick() instead of .omit() for better schema definition in postEvent

mendes-develop committed:
Fix issues identified in PR #63 review

- Fix .coderabbit.yaml parsing error by removing duplicate keys
- Fix CalendarModal logic for showing Edit/Create titles
- Add safety check for event.id in delete operation
- Update EventTable.getEventsByBusinessId to return only id and title as per contract
- Update InsertEventSchema to properly handle nullable startTime and endTime fields

mendes-develop committed:
Fix useCalendarForm hook to properly update when events prop changes

- Update useCalendarForm to return the form object
- Add useEffect to reset form when events prop changes
- Ensure default values for StartDate and EndDate are always valid Date objects

mendes-develop committed:
Add default value for agreeTerms field

- Set default value to false for agreeTerms field in authTable schema

mendes-develop commented:
Acabei de aplicar várias correções com base nos comentários do CodeRabbit e nas solicitações de revisão:

1. Corrigido erro de parsing no arquivo .coderabbit.yaml (chaves duplicadas)
2. Corrigido lógica no CalendarModal.tsx para mostrar corretamente 'Editar' vs 'Criar' evento
3. Adicionado check de segurança para event.id na operação de delete
4. Atualizado EventTable.getEventsByBusinessId para retornar apenas os campos id e title como especificado no contrato
5. Corrigido o hook useCalendarForm para atualizar corretamente quando a propriedade events muda
6. Adicionado valor padrão para o campo agreeTerms no schema

Todos os testes estão passando. Por favor, reveja novamente.

mendes-develop commented:
Closing as superseded. The calendar events feature (POST/GET endpoints, startTime/endTime fields, schema changes) was delivered via CA-1 (#89) and CA-2 (#90) which are already merged to main. This branch's migrations (0001_chunky_thunderball.sql) would corrupt the migration journal by attempting to rename a column (time → startTime) that main has already renamed via a different migration. Any unique UI improvements from this PR can be brought in via a fresh branch off main if needed.
